### DISCUSSION TITLE
How can I use Django-Filters with the POST method?

### DISCUSSION THREAD

**Discussion by skoomer** [carltongibson/django-filter#1472] [Q&A]
How can I use Django-Filters with the POST method?
Is it possible to use this library with the POST method?
And how can this be implemented?

**Comment by carltongibson** [comment]
Normally you instantiate the `FilterSet` with `request.GET`: 

```
def product_list(request):
    f = ProductFilter(request.GET, queryset=Product.objects.all())
    return render(request, 'my_app/template.html', {'filter': f})
```

([View example from the usage guide](https://django-filter.readthedocs.io/en/stable/guide/usage.html#the-view))

Just do the same with `request.POST`: 

```
def product_list(request):
    f = ProductFilter(request.POST, queryset=Product.objects.all())
    return render(request, 'my_app/template.html', {'filter': f})
```

**Reply by skoomer** [reply]
> Normally you instantiate the `FilterSet` with `request.GET`:
> 
> ```
> def product_list(request):
>     f = ProductFilter(request.GET, queryset=Product.objects.all())
>     return render(request, 'my_app/template.html', {'filter': f})
> ```
> 
> ([View example from the usage guide](https://django-filter.readthedocs.io/en/stable/guide/usage.html#the-view))
> 
> Just do the same with `request.POST`:
> 
> ```
> def product_list(request):
>     f = ProductFilter(request.POST, queryset=Product.objects.all())
>     return render(request, 'my_app/template.html', {'filter': f})
> ```

Thank you.
Tell me how to use Django-Filter + jQuery?(ajax)
I use CBV (View) in the POST method I define for example
```
def post(self, request):
    .....
    filter_test = ItemFilter(request.POST, queryset=categories)

    select_check = self.request.POST.getlist('checkBOX[]') where checkBOX its variable from jq
    ......
    context['filter_test'] = filter_test
```

and filters.py

```
class ItemFilter(django_filters.FilterSet):
  category__name = django_filters.ModelMultipleChoiceFilter(field_name='category__name', widget=forms.CheckboxSelectMultiple,queryset=Item.objects.all())
  class Meta:
      model = Item
      fields = ['category__name']
```
I was looking for a long time on Google, but I did not find a good example for an explanation, but I did not find this in the documentation
